### PR TITLE
Explicit String.at/2 access cost and alternatives

### PR DESCRIPTION
**Pull request by sabiwara**
A minor proposal inspired by the explanation in https://hexdocs.pm/elixir/String.html#split_at/2 to improve discoverability.

### CODE REVIEW

File: lib/elixir/lib/string.ex [hunk 1]
```diff
@@ -2231,6 +2231,10 @@ defmodule String do
   Returns the grapheme at the `position` of the given UTF-8 `string`.
   If `position` is greater than `string` length, then it returns `nil`.
 
+  Note: keep in mind this function has to linearly traverse the string.
+  If you want to access a string or a binary in constant time based on the
+  number of bytes, use `Kernel.binary_slice/3` or `:binary.at/2` instead.
```
Comment (whatyouhide) [inline lib/elixir/lib/string.ex:2236]:
```suggestion
  > #### Linear Access {: .warning}
  >
  > This function has to linearly traverse the string.
  > If you want to access a string or a binary in constant time based on the
  > number of bytes, use `Kernel.binary_slice/3` or `:binary.at/2` instead.
```

**Review by whatyouhide — APPROVED** [review]
Think this works better as an admonition block but I’m good with everything 🙃

File: lib/elixir/lib/string.ex (hunk 1 above)
Comment (sabiwara) [reply lib/elixir/lib/string.ex:2236]: I agree this feels like a good use case.
I should also update https://hexdocs.pm/elixir/String.html#split_at/2 I think.

**Comment by sabiwara** [comment]
@whatyouhide I think it looks good:
Screenshot 2024-09-19 at 15 57 07
Screenshot 2024-09-19 at 16 01 18

**Review by josevalim — APPROVED** [review]

**Review by whatyouhide — APPROVED** [review]
Yes, lovely with the admonition blocks!

**Pull request merged**